src/mysql_protocol.erl

```diff
@@ -72,7 +72,8 @@ handshake_finish_or_switch_auth(Handshake, Password, SockModule, Socket,
     {ok, ConfirmPacket, SeqNum1} = recv_packet(SockModule, Socket, SeqNum0),
     case parse_handshake_confirm(ConfirmPacket) of
         #ok{status = OkStatus} ->
-            OkStatus = Handshake#handshake.status,
+            true = (OkStatus == Handshake#handshake.status)
+                    orelse (OkStatus == (16#4000 bor Handshake#handshake.status)),
```

Comment:
1. Please add a comment before these two lines. Suggestion:

```erlang
            %% check status, ignoring bit 16#4000, SERVER_SESSION_STATE_CHANGED.
```

2. Why not
```erlang
            true = OkStatus band bnot 16#4000 == Handshake#handshake.status,
```

I think it is more ideomatic to my understanding of it: _match, ignoring bit 16#4000" in OkStatus_

Reply:
@zuiderkwast Cannot match a record field in Erlang. I updated the PR.

```
ERLC   mysql_protocol.erl
src/mysql_protocol.erl:77: illegal pattern
make[1]: *** [ebin/mysql.app] Error 1
make: *** [app] Error 2
```